x-pack/dockerlogbeat/pipelinemanager/pipelineManager.go

```diff
@@ -62,13 +73,16 @@ func (pm *PipelineManager) CloseClientWithFile(file string) error {
 
 	hash := cl.pipelineHash
 
+	// remove the logger
+	pm.removeLogger(cl.ContainerMeta)
```

Comment:
Is there a chance we have concurrent access to the logger due to `docker logs -f` still being run? Would we need to introduce ref-counting to track all clients that access a log concurrently for cleaning up?

Reply:
So, I've been testing this...and it seems to work? If you spawn multiple `docker logs -f` instances, they'll shut down without any apparent issues once the container is stopped.

Reply:
Nice. Sounds like docker is handling this for us.